docs/scripts/get_mode_requirements.py

```diff
@@ -0,0 +1,394 @@
+                elif flight_mode == 'NAVIGATION_STATE_AUTO_LOITER':
+                    mode_heading_markdown = f"\n### [Hold Mode](../flight_modes_mc/hold.md) ({flight_mode})\n\n"
+                elif flight_mode == 'NAVIGATION_STATE_AUTO_RTL':
+                    mode_heading_markdown = f"\n### [Return Mode](../flight_modes_mc/return.md) ({flight_mode})\n\n"
+                elif flight_mode == 'NAVIGATION_STATE_AUTO_TAKEOFF':
+                    mode_heading_markdown = f"\n### [Takeoff Mode](../flight_modes_mc/takeoff.md) ({flight_mode})\n\n"
+
+                    
+
+
+            elif vehicle_type == 'VEHICLE_TYPE_FIXED_WING':
+                if flight_mode == 'NAVIGATION_STATE_AUTO_MISSION':
+                    mode_heading_markdown = f"\n### [Mission Mode](../flight_modes_fw/mission.md) ({flight_mode})\n\n"
+                elif flight_mode == 'NAVIGATION_STATE_AUTO_LOITER':
+                    mode_heading_markdown = f"\n### [Hold Mode](../flight_modes_fw/hold.md) ({flight_mode})\n\n"
+                elif flight_mode == 'NAVIGATION_STATE_AUTO_RTL':
+                    mode_heading_markdown = f"\n### [Return Mode](../flight_modes_fw/return.md) ({flight_mode})\n\n"          
+                elif flight_mode == 'NAVIGATION_STATE_AUTO_TAKEOFF':
+                    mode_heading_markdown = f"\n### [Takeoff Mode](../flight_modes_fw/takeoff.md) ({flight_mode})\n\n"
+
+
+
+            mode_requirements_markdown += mode_heading_markdown
+
+            for requirement in vehicle_modes_sorted[vehicle_type][flight_mode]:
+                #mode_requirements_markdown += f"- {requirement}: {requirement_defns[requirement]}\n"
+                mode_requirements_markdown += f"- {requirement}\n"
+
+
+    # 2. Define the filename for the overview topic: mode_requirements
+    filename = "mode_requirements.md"
+    filename = docs_output_path_flight_modes / filename
+    try:
+        with open(filename, 'w', encoding='utf-8') as f:
+            # 'w' stands for 'write' mode, which creates the file or overwrites an existing one.
+            f.write(mode_requirements_markdown)
+        
+        print(f"✅ Successfully wrote content to {filename}")
+
+    except Exception as e:
+        print(f"❌ An error occurred: {e}")
+
+
+    # Generate per-doc mode requirements
+    invalid_modes = ['mode_requirements_fixed_wing_position_slow.md',
+                     'mode_requirements_fixed_wing_auto_vtol_takeoff.md',
+                     'mode_requirements_fixed_wing_altitude_cruise.md',
+                     'mode_requirements_fixed_wing_termination.md',
+                     
+                     'mode_requirements_rotary_wing_auto_precland.md',
+                     'mode_requirements_rotary_wing_auto_vtol_takeoff.md',
+                     'mode_requirements_rotary_wing_descend.md',
+                     'mode_requirements_rotary_wing_manual.md',
+                     'mode_requirements_rotary_wing_termination.md' ]
```

Comment:
These don't seem to map to "real modes" that we have documented. Though perhaps precland is, and we should add that requirement.